### DISCUSSION TITLE
CC: Tweaked 1.114 for 1.20.1 and 1.21.1

### DISCUSSION THREAD

**Discussion by SquidDev** [cc-tweaked/CC-Tweaked#2008] [Announcements]
In the past, each Christmas (on ComputerCraft's birthday), I'd try to do a CC:T release. Turns out that's a stupid tradition — even if you've got the release post written beforehand, it still takes up too much time to make a release! Last year I thought I'd try to do something different, and the end-of-year release on CC:T's birthday instead, 15th November. That's today! Yes, it's CC: Tweaked's 7th birthday :tada: :cake:. 

Normally I'd try to keep these posts upbeat, but I think we can all agree it's been a supremely shitty year. I don't really know what to write here which doesn't sound fascile, so just stay safe out there :heart:.

Sorry for my chuntering, let's get onto the actual changes:

## Redstone Relay
Occasionally I like to sit down and play Minecraft with *just* CC:T installed. And while it's surprisingly fun, I always find myself missing one thing — the ability to interact with lots of redstone sources. Lots of other peripheral mods have had this functionality for years (e.g. [Plethora's redstone integrator](https://plethora.madefor.cc/items/redstone-integrator.html)), but this is one of the features that really deserves to be in the base mod itself.

I started work on this way back in 2022, but was held up by texture work. They were [bad](https://github.com/user-attachments/assets/45bc8960-abfa-49b7-8f61-300e4d0bbed4). [Real bad](https://github.com/user-attachments/assets/a9ca99ba-a3ea-46d6-b03a-aee53c69dca8). Thankfully TheStraying11 came to the rescue in #1624, and provided their own textures, and so now CC:T has its own Redstone ~~Integrator~~ Relay!

## Minecraft 1.21.3
Minecraft 1.21.3 came out last month, and CC: Tweaked has been updated to it. Well, sort of — the code is available on the `mc-1.21.y` branch if people want to play with it, but I currently have no plans to make an official release for this version. There's several breaking changes needed for the API to unlock future features (i.e. #1912, #1406), and I'm hoping to get those sorted first.

## In closing
Thanks for everyone who has contributed in the last year :heart:.

**Comment by 80664** [comment]
can we get a discord server

**Reply by Lupus590** [reply]
There's the MineCraft Computer Mods discord which is unofficial but defacto discord for CC.

https://discord.computercraft.cc/ and https://discord.gg/H2UyJXe should both link to it.

**Comment by SirEndii** [comment]
Hey @SquidDev 
I am a bit confused right now. I got a question from an AP user that they can't find a version newer than 1.113.1.
I saw that 1.114.x exists on modrinth and github, but not on curseforge. Is there a reason behind that?
![image](https://github.com/user-attachments/assets/154188b3-68ea-4e78-8f53-7bb39c9095a6)

**Reply by fatboychummy** [reply]
CC:T is no longer published to CF, Squid stated in another discussion that he doesn't wish to support it anymore.

**Reply by SirEndii** [reply]
Oh 
I can understand that

**Reply by SquidDev** [reply]
> Squid stated in another discussion

https://github.com/cc-tweaked/CC-Tweaked/discussions/2017 and 1d7d8006d4edc6db09e0dda9f3bd537492f9d3e3 for reference.

The main CF page does have a whacking big message about this, but if you're only looking at the version page, I guess that's not very helpful :).

**Reply by SirEndii** [reply]
Whoops
Used to just go into the versions
